runtime/wasmer.go

```diff
@@ -127,7 +127,9 @@ func (r *Runtime) StorageRoot() (common.Hash, error) {
 }
 
 func (r *Runtime) Store(data []byte, location int32) {
+	r.mutex.Lock()
 	mem := r.vm.Memory.Data()
+	r.mutex.Unlock()
```

Comment:
This should come after the copy. We should also update `Load`.

Please also make sure to use these functions where possible.

Reply:
when I moved it to after the copy, it causes a deadlock in one of the tests :/ I think we should remove this for now, and I'll open an issue to look into it

Reply:
Yeah you might want to checkout how the lock is actually used...

Reply:
okay I don't think I should change this right now, because it would mean moving the locks in Exec, or rewriting Exec not to use Store. Exec should be an atomic operation (which it is right now), so I don't think it should be changed